generator/src/main/java/com/scottlogic/deg/generator/GenerateExecute.java

```diff
@@ -30,6 +30,14 @@ public GenerateExecute(GenerationConfig config,
 
     @Override
     public void run() {
+
+        if (config.getDataGenerationType() == GenerationConfig.DataGenerationType.RANDOM
+            && config.getMaxRows() == GenerationConfig.Constants.DEFAULT_MAX_ROWS) {
```

Comment:
This shouldn't be checked like this, we need to check if the -n switch was provided.
The generator will not permit the generation of 10,000,000 (`GenerationConfig.Constants.DEFAULT_MAX_ROWS`) rows of random data with this implementation.

```suggestion
            && config.getMaxRows() == null) {
```

And some other means of getting the provided or default max-rows count